sdv/constraints/tabular.py

```diff
@@ -211,78 +214,109 @@ class GreaterThan(Constraint):
+        """
+        if isinstance(scalar_column, list):
+            raise TypeError(f'{scalar} cannot be a list when scalar="{scalar}".')
+
+        column_names = [column_names] if not isinstance(column_names, list) else column_names
+        return column_names
+
+    @classmethod
+    def _validate_inputs(cls, low, high, scalar):
+        if scalar is None:
+            low = [low] if not isinstance(low, list) else low
+            high = [high] if not isinstance(high, list) else high
+            if len(low) > 1 and len(high) > 1:
+                raise ValueError('either `high` or `low` must contain only one column.')
+
+            constraint_columns = tuple(low + high)
+
+        elif scalar == 'low':
+            high = cls._validate_scalar(scalar_column=low, column_names=high, scalar=scalar)
+            constraint_columns = tuple(high)
+        elif scalar == 'high':
+            low = cls._validate_scalar(scalar_column=high, column_names=low, scalar=scalar)
+            constraint_columns = tuple(low)
+        else:
+            raise ValueError(f"scalar {scalar} is unknown. Use either: 'high', 'low', or None.")
 
-        return None
+        return low, high, constraint_columns
 
-    def _get_column_to_reconstruct(self):
+    def _get_columns_to_reconstruct(self):
         if self._drop == 'high':
             column = self._high
         elif self._drop == 'low':
             column = self._low
-        elif self._high_is_scalar:
+        elif self._scalar == 'high':
             column = self._low
         else:
             column = self._high
 
         return column
 
-    def _get_diff_column_name(self, table_data):
-        token = '#'
-        if len(self.constraint_columns) == 1:
-            name = self.constraint_columns[0] + token
-            while name in table_data.columns:
-                name += '#'
+    def __init__(self, low, high, strict=False, handling_strategy='transform',
+                 fit_columns_model=True, drop=None, scalar=None):
+        self._strict = strict
+        self._drop = drop
+        self._scalar = scalar
+        self._low, self._high, self.constraint_columns = self._validate_inputs(low, high, scalar)
+        self._columns_to_reconstruct = self._get_columns_to_reconstruct()
 
-            return name
+        if strict:
+            self.operator = np.greater
+        else:
+            self.operator = np.greater_equal
 
-        while token.join(self.constraint_columns) in table_data.columns:
-            token += '#'
+        super().__init__(handling_strategy=handling_strategy,
+                         fit_columns_model=fit_columns_model)
+
+    def _get_value(self, table_data, field):
+        variable = getattr(self, f'_{field}')
+        if self._scalar == field:
+            return variable
+        elif all([column in table_data.columns for column in variable]):
+            return table_data[getattr(self, f'_{field}')].values
+
+        return None
```

Comment:
I think this return None is not necessary, and neither is the `elif` above.

This can be:
```
if self._scalar == field:
    return variable

return table_data[variable].values
```

Reply:
In reverse transform L427-428 we try to get the value of high and low:
```
high = self._get_value(table_data, 'high')
low = self._get_value(table_data, 'low')
```

but because of `drop` there is no guarantee these columns exist, raising a `KeyError`.

Reply:
Ok, there's the problem :-)
Adding a comment on L427